### PR TITLE
Convert iconButton to Typescript

### PR DESCRIPTION
**Pull request by eniskraasniqi1**
### Description

Converting Icon Button to typescript.

### CODE REVIEW

File: src/components/iconButton/IconButton.tsx [hunk 1]
```diff
@@ -1,57 +1,89 @@
...
 import Icon from '../icon';
 import cx from 'classnames';
 import buttonTheme from '../button/theme.css';
 import theme from './theme.css';
 import LoadingSpinner from '../loadingSpinner';
+import { BoxProps } from '../box/Box';
+
+type SIZES = 'small' | 'medium' | 'large';
+type TINTS = 'lightest' | 'light' | 'normal' | 'dark' | 'darkest';
+type COLOURS = 'neutral' | 'white' | 'mint' | 'violet' | 'ruby' | 'gold' | 'aqua' | 'teal';
```
Comment (qubis741) [inline src/components/iconButton/IconButton.tsx:12]: COLORS (without U)

File: src/components/iconButton/IconButton.tsx [hunk 2]
```diff
@@ -1,57 +1,89 @@
...
+
+export interface IconButtonProps extends Omit<BoxProps, 'ref'> {
```
Comment (qubis741) [inline src/components/iconButton/IconButton.tsx:14]: all props from here that are passed to `Box` (L 99) needs to be mentioned here, if you are somehow overriding them

Comment (eniskraasniqi1) [reply src/components/iconButton/IconButton.tsx:14]: Missed those 😬 . Adding the overrided ones now.

**Review by qubis741 — APPROVED** [review]

File: src/components/iconButton/iconButton.stories.spec.tsx [hunk 3]
```diff
@@ -8,7 +8,7 @@ export default {
   title: 'IconButton',
 };
 
-const Wrapper = ({ children, inverse = false }) => (
+const Wrapper = ({ children, inverse = false }: { children: ReactNode; inverse: boolean }) => (
```
Comment (JorenSaeyTL) [inline src/components/iconButton/iconButton.stories.spec.tsx:11]: PP: `interface WrapperProps` and no inline typing

**Review by JorenSaeyTL — APPROVED** [review]

**Review by lowiebenoot — APPROVED** [review]

**Comment by qubis741** [comment]
Manually checking :hourglass:

**Comment by qubis741** [comment]
Manually checked :white_check_mark:

**Pull request merged**